test/letters/01-authed.e2e.spec.js

```diff
@@ -0,0 +1,65 @@
+const E2eHelpers = require('../e2e/e2e-helpers');
+const Timeouts = require('../e2e/timeouts.js');
+const LettersHelpers = require('../e2e/letters-helpers.js');
+const LoginHelpers = require('../e2e/login-helpers');
+
+module.exports = E2eHelpers.createE2eTest(
+  (client) => {
+    const token = LoginHelpers.getUserToken();
+
+    LettersHelpers.initApplicationMock(token);
+
+    // Ensure main status page renders.
+    LoginHelpers.logIn(token, client, '/letters', 3)
+      .waitForElementVisible('body', Timeouts.normal)
+      .axeCheck('.main')
+      .assert.title('Download VA Letters: Vets.gov')
+      .waitForElementVisible('.letters', Timeouts.slow);  // First render of React may be slow.
+
+    client.axeCheck('.main');
+
+    // Checking if full name has rendered
+    client.expect.element('h5.letters-address').text.to.contain('William Shakespeare');
+
+    // Checking if address has rendered correctly
+    client.expect.element('.street').text.to.contain('57 Columbus Strassa, Ben Franklin Village');
+    client.expect.element('.city-state').text.to.contain('APO, AE 09028');
+
+    // Update address and cancel
+    client
+      .click('.usa-button-outline')
+      .expect.element('select').to.be.present;
+
+    client
+      .setValue('select[name="country"]', 'United Kingdom')
+      .clearValue('input[name="city"]')
+      .fill('input[name="city"]', 'Stratford-upon-Avon')
```

Comment:
I was just reading MacBeth the other day, so this makes me probably happier than it should. 😄

Reply:
💯